src/WebJobs.Script/Binding/FunctionBinding.cs

```diff
@@ -133,5 +136,60 @@ internal static void AddStorageAccountAttribute(Collection<CustomAttributeBuilde
             var attribute = new CustomAttributeBuilder(typeof(StorageAccountAttribute).GetConstructor(constructorTypes), constructorArguments);
             attributes.Add(attribute);
         }
+
+        protected static ICollection<JToken> ReadAsCollection(Stream valueStream)
+        {
+            byte[] bytes;
+            using (MemoryStream ms = new MemoryStream())
+            {
+                valueStream.CopyTo(ms);
+                bytes = ms.ToArray();
+            }
+
+            JArray values = null;
+            string stringValue = Encoding.UTF8.GetString(bytes);
+
+            if (Utility.IsJson(stringValue))
+            {
+                JToken token = JToken.Parse(stringValue);
+                values = token as JArray;
+                if (token.Type != JTokenType.Array)
+                {
+                    values = new JArray();
+                    values.Add(token);
+                }
+            }
+            else
+            {
+                values = new JArray();
+                values.Add(stringValue);
+            }
+
+            return values;
+        }
+
+        protected async Task BindAsyncCollectorAsync<T>(Stream stream, IBinderEx binder, RuntimeBindingContext runtimeContext)
+        {
+            IAsyncCollector<T> collector = await binder.BindAsync<IAsyncCollector<T>>(runtimeContext);
+            ICollection<JToken> values = ReadAsCollection(stream);
+            foreach (var value in values)
+            {
+                object converted = null;
+                if (typeof(T) == typeof(string))
+                {
+                    converted = value.ToString();
+                }
+                else if (typeof(T) == typeof(JObject))
+                {
+                    converted = (JObject)value;
+                }
+                else
+                {
+                    throw new ArgumentException("Unsupported collection type.");
+                }
+
```

Comment:
I have tests for the array of string case since that's one of the primary supported scenarios, but you're right that the array of arrays isn't supported right now. I'll add a note to improve error handling for this.